frontend/src/PublishedPostsTile.tsx

```diff
@@ -108,56 +118,60 @@ export class PublishedPostsTile extends React.Component<PublishedPostsProps, Pub
                     ToastManager.showErrorToast(response);
-                    this.state.eventBus.fireEvent(EventBusEventType.REFRESH_POSTS);
+                    this.props.eventBus.fireEvent(EventBusEventType.REFRESH_POSTS);
                 } else {
-                    ToastManager.showSuccessToast("Created and immediately published post.");
+                    ToastManager.showSuccessToast("Published post.");
                     response.json().then(data => {
-                        this.state.eventBus.fireEvent(EventBusEventType.REFRESH_POSTS);
+                        this.props.eventBus.fireEvent(EventBusEventType.REFRESH_POSTS);
                     });
                 }
             });
     }
 
     public render() {
+        //rows with posts
         const posts = this.state.posts.sort(
             (p1: PublishedPost, p2: PublishedPost) => p2.created.getTime() - p1.created.getTime())
             .map(elem => {
+                //status: first column with checkmark or error with error description.
                 var status;
                 if (elem.errorMessage) {
-                    status = <td className="tooltip centered icon-medium">{ImageProvider.getImage("none-logo")}
+                    status = <td key={elem.id} className="tooltip centered icon-medium">{ImageProvider.getImage("none-logo")}
                         <span className="tooltiptext">{elem.errorMessage}</span></td>
 
                 } else {
                     if (elem.postUrl) {
-                        status = <td onClick={o => this.goToPost(elem)} className="checkmark centered pointer">{ImageProvider.getImage("check")}{ImageProvider.getImage("link")}</td>
+                        status = <td key={elem.id} onClick={o => this.goToPost(elem)} className="checkmark centered pointer">{ImageProvider.getImage("check")}{ImageProvider.getImage("linked")}</td>
                     } else {
                         const statusMsg = elem.scheduled ? "This post is scheduled for " + moment(elem.scheduled).format("YYYY-MM-DD HH:mm") : "This post is neither published nor scheduled yet.";
                         const statusIcons = elem.scheduled ? [ImageProvider.getImage("check"), ImageProvider.getImage("clock-small")] : ImageProvider.getImage("check");
-                        status = <td className="checkmark checkmark-gray centered tooltip">{statusIcons}<span className="tooltiptext">{statusMsg}</span></td>
+                        status = <td key={elem.id} className="checkmark checkmark-gray centered tooltip">{statusIcons}<span className="tooltiptext">{statusMsg}</span></td>
                     }
                 }
 
-
+                //actions for the current post
                 const allowedActions = (
                     <div className="inline-block">
                         <span className="table-icon">
                             <span onClick={() => this.selectPost(elem)}>{ImageProvider.getImage("edit")}
                             </span>
```

Comment:
Looks a bit weird with a new line but no indention because children are in the same line as the parent opening but not the closing tag.
```suggestion
                            <span onClick={() => this.selectPost(elem)}>
                                {ImageProvider.getImage("edit")}
                            </span>
```